src/components/validation/validators/component_spec/sources.rs

```diff
@@ -398,3 +402,71 @@ fn validate_component_sent_event_bytes_total(
         metric_bytes,
     )])
 }
+fn validate_component_errors_total(
+    _configuration: &ValidationConfiguration,
+    inputs: &[TestEvent],
+    _outputs: &[Event],
+    telemetry_events: &[Event],
+) -> Result<Vec<String>, Vec<String>> {
+    let mut errs: Vec<String> = Vec::new();
+
+    let metrics = filter_events_by_metric_and_component(
+        telemetry_events,
+        &SourceMetrics::ErrorsTotal,
+        TEST_SOURCE_NAME,
+    );
+
+    let mut errors: u32 = 0;
+    for m in metrics {
+        match m.value() {
+            vector_core::event::MetricValue::Counter { value } => {
+                if let MetricKind::Absolute = m.data().kind {
+                    errors = *value as u32
+                } else {
+                    errors += *value as u32
+                }
+            }
+            _ => errs.push(format!(
+                "{}: metric value is not a counter",
+                SourceMetrics::ErrorsTotal,
+            )),
+        }
+    }
```

Comment:
non-blocking: I noticed this pattern existing in a number of other areas of the validation code. I wonder if it could be extracted for a helper function.

Reply:
Thanks, I created a helper here.